bundles/mapping/drawtools/plugin/DrawPlugin.ol.js

```diff
@@ -928,11 +928,21 @@ Oskari.clazz.define(
                 // intersection is allowed or geometry isn't being drawn currently
                 return;
             }
-            if (geometry.getCoordinates()[0].length < 4 || !isValidOp.isValid(olParser.read(geometry))) {
-                // lines intersect -> problem!!
-                currentDrawing.setStyle(me._styles.intersect);
-                me._featuresValidity[currentDrawing.getId()] = false;
-                return;
+
+            if (geometry.getCoordinates()[0].length >= 4) {
+                // This function is called twice when modifying geometry from the point where the drawing initially began
+                // That is because the first and the last point of the geometry are being modified at the same time (they should have identical values)
+                // The first call causes an error because the points differ from each other. Hence using try/catch here to avoid breaking the js process
+                try {
+                    if (!isValidOp.isValid(olParser.read(geometry))) {
+                        // lines intersect -> problem!!
+                        currentDrawing.setStyle(me._styles.intersect);
+                        me._featuresValidity[currentDrawing.getId()] = false;
+                        return;
+                    }
+                } catch (e) {
+                    return;
```

Comment:
It would be better to prevent the exception if we know it's going to happen instead of try-catch.